package.json

```diff
@@ -35,27 +36,58 @@
     "fix": "npm run fix:lint && npm run fix:prettier",
     "start": "npm run develop",
     "serve": "gatsby serve",
-    "test": "echo \"Write tests! -> https://gatsby.app/unit-testing\"",
-    "commit": "git-cz"
+    "commit": "git-cz",
+    "cypress:open": "cross-env TZ=UTC node_modules/.bin/cypress open",
+    "cypress:run": "cross-env TZ=UTC node_modules/.bin/cypress run",
+    "test": "cross-env NODE_ENV=test npm run test:unit && npm run test:e2e",
+    "test:unit": "nyc --silent --no-clean jest",
+    "test:e2e": "nyc --silent --no-clean start-server-and-test develop http://localhost:8000 cypress:run",
+    "pretest": "chmod +x scripts/**.* && cross-env ./scripts/pre-test.sh",
+    "posttest": "cross-env ./scripts/build_report.sh"
   },
   "devDependencies": {
+    "@cypress/code-coverage": "^1.10.2",
     "@typescript-eslint/eslint-plugin": "2.7.0",
     "@typescript-eslint/parser": "2.7.0",
     "babel-eslint": "10.0.3",
     "commitizen": "^4.0.3",
-    "cssnano": "^4.1.10",
     "cz-conventional-changelog": "^3.0.2",
+    "babel-jest": "^24.9.0",
+    "babel-plugin-istanbul": "^5.2.0",
+    "babel-preset-gatsby": "^0.2.23",
+    "chai": "^4.2.0",
+    "chai-enzyme": "^1.0.0-beta.1",
+    "coveralls": "^3.0.9",
+    "cross-env": "^6.0.3",
+    "cssnano": "^4.1.10",
+    "cypress": "^3.7.0",
+    "enzyme": "^3.10.0",
+    "enzyme-adapter-react-16": "^1.15.1",
     "eslint": "^6.6.0",
     "eslint-config-react-app": "5.0.2",
+    "eslint-plugin-chai-friendly": "^0.5.0",
+    "eslint-plugin-cypress": "^2.7.0",
     "eslint-plugin-flowtype": "4.4.1",
     "eslint-plugin-import": "2.18.2",
     "eslint-plugin-jsx-a11y": "6.2.3",
     "eslint-plugin-react": "7.16.0",
     "eslint-plugin-react-hooks": "2.3.0",
+    "identity-obj-proxy": "^3.0.0",
+    "istanbul-instrumenter-loader": "^3.0.1",
+    "istanbul-lib-coverage": "^2.0.5",
+    "jest": "^24.9.0",
+    "jest-environment-enzyme": "^7.1.2",
+    "jest-enzyme": "^7.1.2",
+    "nyc": "^14.1.1",
     "prettier": "1.19.1",
+    "shx": "^0.3.2",
+    "start-server-and-test": "^1.10.6",
```

Comment:
I fear these dependencies could be extraneous. 

I'll explain why:

* `shx` is a wrapper around ShellJS Unix commands, providing an easy solution for simple Unix-like, cross-platform commands in NPM package scripts. Meaning it's meant to enable you to run `unix-like` commands in javascript land, but I'm seeing you're making use of it in an actual shell script which isn't really needed.
Can you clarify if there's a need for it apart from my assumption?

* For `start-server-and-test`, I'm not exactly sure about this but I'm thinking if what it does is similar to what the `cypress` executable already provides for us.

Again, this is just a suggestion and I'm trying to understand.

Reply:
<img width="602" alt="Screenshot 2019-12-02 at 3 45 12 PM" src="https://user-images.githubusercontent.com/29008971/70068194-d0a40a80-15ef-11ea-9816-06df5958149c.png">
`Cypress` needs the server running inother to test and `start-server-and-test` handle this very well.

Reply:
Cypress needs a running instance of the application to run `end-to-end` tests, yes, but since for every PR there's a PR-deploy based on netlify, why don't you make use of that URL for testing.

Creating a running instance just for the sole purpose of `end-to-end` may be an unnecessary step.
@BolajiAyodeji kindly clarify what the process for this looks like.

Reply:
You have a point there.

Reply:
@BolajiAyodeji is it possible to make use of the URL of the deployed branch when running `end-to-end` tests instead of starting a running instance of the application on whatever CI provider we are using. This will make the tests run faster I think.